third_party/java/arcs/build_defs/internal/schemas.bzl

```diff
@@ -164,16 +159,110 @@ def arcs_kt_schema(
         )
     return {"outs": outs, "deps": outdeps}
 
+def _arcs_ts_preproc_impl(ctx):
+    outputs = [
+        ctx.actions.declare_file(x.basename, sibling = x)
+        for x in ctx.files.srcs
+    ]
+
+    ctx.actions.run_shell(
+        inputs = ctx.files.srcs,
+        outputs = outputs,
+        arguments = [src.path for src in ctx.files.srcs],
+        command = """
+        for i in "$@"; do
+          out="$(pwd)/bazel-out/host/bin/$i"
+          sed -e 's/-web.js/-node.js/g' $i > "$out"
```

Comment:
eventually we will want to build the entire web version in bazel as well, perhaps there should be an enable/disable flag (default platform, node)